contracts/src/v0.8/mocks/VRFCoordinatorV2Mock.sol

```diff
@@ -85,19 +85,36 @@ contract VRFCoordinatorV2Mock is VRFCoordinatorV2Interface {
    * @param _consumer the VRF randomness consumer to send the result to
    */
   function fulfillRandomWords(uint256 _requestId, address _consumer) external {
+    fulfillRandomWordsWithOverride(_requestId, _consumer, new uint256[](0));
+  }
+
+  /**
+   * @notice fulfillRandomWordsWithOverride allows the user to pass in their own random words.
+   *
+   * @param _requestId the request to fulfill
+   * @param _consumer the VRF randomness consumer to send the result to
+   * @param _words user-provided random words
+   */
+  function fulfillRandomWordsWithOverride(
+    uint256 _requestId,
+    address _consumer,
+    uint256[] memory _words
+  ) public {
     uint256 startGas = gasleft();
     if (s_requests[_requestId].subId == 0) {
       revert("nonexistent request");
     }
     Request memory req = s_requests[_requestId];
 
-    uint256[] memory words = new uint256[](req.numWords);
-    for (uint256 i = 0; i < req.numWords; i++) {
-      words[i] = uint256(keccak256(abi.encode(_requestId, i)));
+    if (_words.length != req.numWords) {
```

Comment:
Minor nit: I'm not sure if this condition is good enough - I can imagine someone messing up their mocking and providing 2 words instead of 3, and their test failing in an odd way. Would it be better to just check if `_words.length == 0` and populate it then, and if `_words.length != req.numWords`, revert with an error like `InvalidMockedRandomWords`? 

Without knowing too much about what would be preferred by users, I can't be too sure, but I think might be better to be as obvious as possible to minimize time debugging mocks in tests.

Reply:
I think that's a good change. In that case I'll also add a test, because at this point there's a bit of added logic.